### ISSUE TITLE
Cannot load Analyzers programatically from MSBuildWorkspace with Microsoft.CodeAnalysis 1.1.1

### ISSUE THREAD

**Issue by yaakov-h** [dotnet/roslyn#7654]
Scenario:

1. Create a new .NET 4.6 command-line project
2. Add `Microsoft.CodeAnalysis` 1.1.1 and [StyleCop.Analyzers 1.0.0-rc2](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) to a project with NuGet
3. Project now has 5 Analyzer references:
  1. Microsoft.CodeAnalysis.Analyzers
  2. Microsoft.CodeAnalysis.CSharp.Analyzers
  3. Newtonsoft.Json (empty)
  4. StyleCop.Analyzers
  5. StyleCop.Analyzers.CodeFixes
4. Add the following code to the project and run it:
```
var workspace = MSBuildWorkspace.Create();
var project = await workspace.OpenProjectAsync(projectPath);

var allLanguageAnalyzers = project.AnalyzerReferences.SelectMany(r => r.GetAnalyzersForAllLanguages());
var languageSpecificAnalyzers = project.AnalyzerReferences.SelectMany(r => r.GetAnalyzers(project.Language));
```
  Where `projectPath` is the path to this project.

Expectations:
* `allLanguageAnalyzers` is not empty
* `languageSpecificAnalyzers` is not empty

Actual Results:
* `allLanguageAnalyzers` is empty
* `languageSpecificAnalyzers` is empty

Note:
* When using Microsoft.CodeAnalysis `1.0.0` instead of `1.1.1`, this all works. I have attempted `1.1.1` with the following assembly redirects, but to no avail.
* The internal exception seems to be a `FileNotFoundException` which is caught.
* Even without adding StyleCop, the Microsoft.CodeAnalysis analyzers are not returned on 1.1.1. This I find exceptionally odd.

```
  <runtime>
    <assemblyBinding>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.VisualBasic" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.VisualBasic.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces.Desktop" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.27.0-1.1.65535.65535" newVersion="1.1.37.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Composition.AttributedModel" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.65535.65535" newVersion="1.0.27.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Composition.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.65535.65535" newVersion="1.0.27.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.99.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```

**Comment by jmarolf** [comment]
@yaakov-h are you using VS 2015 RTM or Update 1?

**Comment by yaakov-h** [comment]
Update 1.

**Comment by jmarolf** [comment]
Looks like a problem in the Update 1 assemblies.  I'll try and get a fix out today.

**Comment by jmarolf** [comment]
@yaakov-h add `xmlns="urn:schemas-microsoft-com:asm.v1` to you `assemblyBinding` definition in `App.config`
 ```xml
 <runtime>
    <assemblyBinding  xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.VisualBasic" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.VisualBasic.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces.Desktop" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
        <bindingRedirect oldVersion="0.7.0.0-1.1.0.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.27.0-1.1.65535.65535" newVersion="1.1.37.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Composition.AttributedModel" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.65535.65535" newVersion="1.0.27.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Composition.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.65535.65535" newVersion="1.0.27.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
        <bindingRedirect oldVersion="1.0.0.0-1.0.99.0" newVersion="1.1.0.0"/>
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```

That will allow the clr to load analyzers that depend on 1.0 binaries.

**Comment by yaakov-h** [comment]
Thanks @jmarolf, that did the trick.

**Comment by shaggygi** [comment]
@jmarolf I was trying to something similar with current Nuget packages.  Seems like it won't read the analyzers and just returns 0.  What are the steps again to fix this?  Thx
